### ISSUE TITLE
Errors are not shown in stand alone script files

### ISSUE THREAD

**Issue by halcwb** [ionide/ionide-vscode-fsharp#1515]
### Describe the bug**
With the latest version of VS and Ionide, errors are not shown in a script file.

### Steps to reproduce
1. Open VS Code (used version etc.. see below).
2. Choose new file and set the language to F#
3. Write some code with obvious errors

### Link to sample reproduction

```fsharp
let xs = [ "a"; 1 ]
```

### Expected behaviour
There should be an error sign under the above code

### Screenshots
![image](https://user-images.githubusercontent.com/683631/111225746-7494a800-85e0-11eb-916f-304c5d78383e.png)

However, no error is shown.

## Machine info
Version: 1.54.2 (system setup)
Commit: fd6f3bce6709b121a895d042d343d71f317d74e7
Date: 2021-03-11T00:56:19.848Z
Electron: 11.3.0
Chrome: 87.0.4280.141
Node.js: 12.18.3
V8: 8.7.220.31-electron.0
OS: Windows_NT x64 6.1.7601

Ionide-fsharp v5.4.0

**Comment by baronfel** [comment]
Important to note here that you're in an un-saved script file. I can't repro this in a saved script file. Our processing of untitled documents in FSAC could have regressed.

**Comment by halcwb** [comment]
@baronfel You're right upon save the error displays. Only I have another script file that won't show the errors. 

![image](https://user-images.githubusercontent.com/683631/111227266-968f2a00-85e2-11eb-9f53-5d281a8f6645.png)

But why in this script file and not the other?? 

One thing to note, this script file is 1000 lines. So, could that be the problem?

**Comment by baronfel** [comment]
I can't tell anything really without the full contents of the file. Do any errors at all appear? Is it only this kind of error that doesn't appear, etc, etc.

**Comment by halcwb** [comment]
@baronfel No errors appear at all, indeed. I know this isn't really a clear reproducible problem. However, when I just delete all proceeding code, still no errors appear:

![image](https://user-images.githubusercontent.com/683631/111231081-e5d85900-85e8-11eb-854e-1bd4524f4c3a.png)

**Comment by nhirschey** [comment]
@halcwb , try 

1. Close the file
2. Close the folder using File > Close folder dialog
3. Reopen folder using the File > open folder (or open recent folder) dialog
4. reopen script file.

This tends to work for me. If this fixes it, then it's probably a similar bug to https://github.com/ionide/ionide-vscode-fsharp/issues/1500

**Comment by halcwb** [comment]
@nhirschey Difficult to say. Unfortunately, the problem is all over the place and seemingly inconsistent. For example, I tried your method without success and as shown below, the error is detected in the file view, but doesn't show in the editor window:

![image](https://user-images.githubusercontent.com/683631/111430692-02fd4c80-86fb-11eb-9450-397762de859e.png)

**Comment by nhirschey** [comment]
That screenshot is of ErrorBug.fs. Is ErrorBug.fsx working now in editor view after following my steps?

**Comment by halcwb** [comment]
@nhirschey Oops, you're right. With small simple script files your solution works. However, with a large script file (including a typeprovider), the error messages never show up, whatever I do.

**Comment by halcwb** [comment]
The frustrating thing is that currently I cannot run the script file using any editor. Visual Studio can't handle net5 libs. Rider, the same and VSCode isn't giving any error warnings (also is quite slow). VS for Mac can't handle script files at all. I tried them all.

**Comment by cartermp** [comment]
Visual Studio supports running a net5 script, and has full IntelliSense for it in the 16.10 previews. You need to turn it on in the settings though (it will be the default in the next major version of VS). Can't speak towards Ionide or Rider at the moment.

**Comment by halcwb** [comment]
@baronfel Your last bullet on the list is the issue at hand. I am struggling to get a repro. It's a long script file using a type provider. I can send you the code file? The strange thing is if I delete all the code in the script file and do a save as, still no errors show in that script file when you type some invalid code.

@cartermp Can I download the preview? Because with the current version with use of net core scripting enabled, it doesn't work.

**Comment by baronfel** [comment]
I guess some clarity is in order: when you say 'run', do you actually mean run? like using the `send to FSI` command in VS code and FSI running the script? or are you meaning that typechecking/editor features don't work for that script? it sounds like the latter, but the phrasing is vague enough that I want to pin down what you mean.

**Comment by halcwb** [comment]
@baronfel It's specifically the error checking. I can indeed run the script. But when I start editing the script I get the intellisense, the code completion but no error detection. I think the screenshots are quite clear.

**Comment by nhirschey** [comment]
@halcwb , see here for Phillip's post about .net core scripting in preview Visual studio and how to enable it: https://devblogs.microsoft.com/dotnet/f-and-f-tools-update-for-visual-studio-16-9/  

The scripts of mine that had trouble in ionide were properly type-checked if I opened them in VS2019 instead, so .net core scripting in VS19 should work. Though now I just follow the close/reopen procedure to get ionide type-checking back whenever it happens. It's not a one-time fix; you may have to do it a couple times on the same file over the span of a couple hours.

**Comment by baronfel** [comment]
@halcwb can you add the schema to that gist as well, or a description of it?

**Comment by nhirschey** [comment]
@baronfel , I don’t know how the tooling works, so this is a long shot but my scripts where typechecking randomly stops working in ionide also reference FSharp.Stats and Plotly.NET. I can’t imagine that matters, but it’s a similarity so I thought I’d mention it.

**Comment by baronfel** [comment]
@nhirschey for a brief explainer, in order to typecheck we need to pieces of data:
* the contents of the file to typecheck, and
* a set of options describing what parameters to use to configure the compiler when checking the file (this includes references, #defines, target framework, etc)

The compiler services have helpers that help generate the options for scripts, and the options for files-in-a-project are derived from the project file.

It could be that in the code we have to handle scripts there's some condition that results in the options getting modified or corrupted, but there's not any good monitoring or operational logging around that at the moment to help us debug.

**Comment by halcwb** [comment]
@baronfel I have attached the two schema files:

[shema files.zip](https://github.com/ionide/ionide-vscode-fsharp/files/6162378/shema.files.zip)

A couple of things to note:

1. Due to work restrictions I used VSCode on a Windows 7 PC. But now I also tried a Windows 10 PC installation of VSCode. And then the bug behaves exactly as described by @nhirschey, with the same work around.
2. Also using Windows 10, VSCode seems a lot faster (PC specs of the Win10 are actually way below the specs of the Win7 PC).

See with that I will close this bug as a duplicate of https://github.com/ionide/ionide-vscode-fsharp/issues/1500.

Thanks for listening in and the help @baronfel @nhirschey @cartermp. Really appreciate it.
